packages/common/src/core/manager.ts

```diff
@@ -0,0 +1,27 @@
+import { AppConfig, ConfigManager, Dependencies } from '../config/config.js';
+import { dictionaryRouters } from '../routers/dictionaryRouter.js';
+
+export class LyricManager extends ConfigManager {
+	public static async create(configData: AppConfig) {
+		const manager = new LyricManager(configData);
+		await manager.loadLogger();
+		await manager.loadDb();
+		return manager;
+	}
+	getRouters() {
+		return {
+			getDicionaryRouters: () => {
+				const routers = dictionaryRouters({
+					db: this.dependencies.db,
+					config: this.dependencies.config,
+					logger: this.dependencies.logger,
+				} as Dependencies);
```

Comment:
rather than asserting "as Type" you can have the function itself return the type, which makes it easier to notice issues within that function if changes are being made whereas the assertion would mask/hide those issues.

Reply:
changed the manager structure and how dependencies object is passed to functions